### ISSUE TITLE
Permission error with newest version: 'iam.serviceAccounts.signBlob' denied on resource

### ISSUE THREAD

**Issue by hendrik244** [strapi-community/strapi-provider-upload-google-cloud-storage#171]
Hi,

just updated to the latest version of strapi and this plugin.
All services are hosted in gcp.
After updating I received this error, when trying to fetch content with images (stored in gcs) included:

<img width="1049" alt="Bildschirmfoto 2023-05-25 um 10 34 47" src="https://github.com/strapi-community/strapi-provider-upload-google-cloud-storage/assets/33331414/3933f873-1367-4a58-8fd6-f70b100be122">

After rolling it back to the former used version (4.5.6) it all runs fine again. Even with the latest strapi version.

Best regards,
Hendrik

**Comment by tjablonski** [comment]
I have same issue with 4.10.2

**Comment by marcneander** [comment]
Same here

**Comment by hendrik244** [comment]
This helped me out: https://stackoverflow.com/questions/75071759/google-cloud-storage-permission-denied

Maybe not the plugin, but gcp is the problem...

**Comment by knokmki612** [comment]
> Maybe not the plugin, but gcp is the problem...

I think it also relate to release version of 4.10.1
https://github.com/strapi-community/strapi-provider-upload-google-cloud-storage/releases/tag/4.10.1

Because this release supports gcs signed url
https://cloud.google.com/storage/docs/access-control/signed-urls#signing-iam

If my guess is correct, enable [publicFiles options](https://github.com/strapi-community/strapi-provider-upload-google-cloud-storage#publicfiles) will be change the situation.

**Comment by LaurenceHo** [comment]
Is that possible we don't use signed url on public files? or maybe we can config whether we want to use signed URL or not?

**Comment by benaduggan** [comment]
> This helped me out: [stackoverflow.com/questions/75071759/google-cloud-storage-permission-denied](https://stackoverflow.com/questions/75071759/google-cloud-storage-permission-denied)
> 
> Maybe not the plugin, but gcp is the problem...

This also helped fix things for me! Specifically adding the "Service Account Token Creator" role to our service account. Also just wanted to show off my config block in case it helps anyone else! 

![image](https://github.com/strapi-community/strapi-provider-upload-google-cloud-storage/assets/5338872/a290006c-ab79-44bb-8906-56440d83681f)

**Comment by knokmki612** [comment]
> Is that possible we don't use signed url on public files?

Yes that's possible. if you set `true` to `publicFiles` option, you can get unsigned url. This behavior documented as follows:

> Files are signed when publicFiles is set to false.

https://github.com/strapi-community/strapi-provider-upload-google-cloud-storage#expires

> or maybe we can config whether we want to use signed URL or not?

We can choose to use signed URL or not by publicFiles option.

**Comment by kuatek** [comment]
This helped me: https://stackoverflow.com/questions/75071759/google-cloud-storage-permission-denied

**Comment by cyp3rius** [comment]
Closing, hence no activity & support only for Strapi 5.
